File: darkSky.py
from darksky.api import DarkSky
from darksky.types import languages, units, weather
from geopy.geocoders import Nominatim

geolocator = Nominatim(user_agent="specify_your_app_name_here")
location = geolocator.geocode("Ciudad de Mexico, Mexico")

API_KEY = '34c1183964f3da3d546e551b2f863648'

darksky = DarkSky(API_KEY)

forecast = darksky.get_forecast(
    location.latitude,
    location.longitude,
    extend=False, # default `False`
    lang=languages.ENGLISH, # default `ENGLISH`
    values_units=units.AUTO, # default `auto`
    exclude=[weather.MINUTELY, weather.ALERTS], # default `[]`,
    #timezone='UTC' # default None - will be set by DarkSky API automatically
)


location = geolocator.reverse(f'{forecast.latitude}, {forecast.longitude}')
print(f'{location.raw["address"]["state"]}, {location.raw["address"]["country"]} = lat: {location.latitude} - lon: {location.longitude} ')
print(f'Temperature: {forecast.currently.temperature}°')
print(f'Current summary: {forecast.currently.summary}')
print(f'Timezone: {forecast.timezone}')

print('''\n  ---------------------------------------------------------------------
      Date      Temperature(avg)              Date summary
  ---------------------------------------------------------------------''')
for day in forecast.daily.data:
    avg = (day.temperature_max + day.temperature_min) / 2
    print(f'   {day.time.date()}        {round(avg, 1)}°           {day.summary}')
print('  ---------------------------------------------------------------------')
print(f'  Forecasted week summary: {forecast.daily.summary}')

File: darkSky.go
package main

import (
	"fmt"
	"reflect"
	"time"

	"github.com/shawntoffel/darksky"
)

func main() {
	client := darksky.New("KEY")

	request := darksky.ForecastRequest{
		Latitude:  19.4327224,
		Longitude: -99.1333089,
		Options: darksky.ForecastRequestOptions{
			Exclude: "hourly,minutely",
			Units:   "auto", // SI = Celsius
		},
	}

	forecast, err := client.Forecast(request)
	if err != nil {
		panic(err)
	}

	e := reflect.ValueOf(&forecast).Elem()

	for i := 0; i < e.NumField(); i++ {
		varName := e.Type().Field(i).Name
		varType := e.Type().Field(i).Type
		varValue := e.Field(i).Interface()
		fmt.Printf("%v %v %v\n", varName, varType, varValue)
	}

	fmt.Printf("Latitud: %v - Longitud: %v\n", forecast.Latitude, forecast.Longitude)
	fmt.Printf("Temperature: %v°\n", forecast.Currently.Temperature)
	fmt.Printf("Current summary: %v\n", forecast.Currently.Summary)
	fmt.Printf("Timezone: %v\n", forecast.Timezone)

	fmt.Printf("Forecasted week summary: %v\n", forecast.Daily.Summary)
	for _, day := range forecast.Daily.Data {
		tm := time.Unix(int64(day.Time), 0) // Converts timestamp to int64. Gets DateTime
		fmt.Println(tm.Date())
		fmt.Println(day.Summary)
		avg := (day.TemperatureMax + day.TemperatureMin) / 2
		fmt.Printf("%.1f°\n", avg) // Rounds to a floating-point value
	}
}
